File: zadanie2/csharp/main.cs
using System;
using System.Collections.Generic;

class Crop
{
    public string Name;
    public int Growth;
    public int YieldAmount;
    public string WeatherImpact;

    public Crop(string name, int growth, int yieldAmount, string weatherImpact)
    {
        Name = name;
        Growth = growth;
        YieldAmount = yieldAmount;
        WeatherImpact = weatherImpact;
    }

    public void Grow(string weatherType)
    {
        if (weatherType == "sunny")
        {
            Growth += 1;
        }
        else if (weatherType == "rainy")
        {
            Growth += 2;
        }
        else if (weatherType == "stormy")
        {
            Growth += 0;
        }
    }

    public int Harvest()
    {
        if (Growth >= 3)
        {
            return YieldAmount;
        }
        else
        {
            return 0;
        }
    }
}

class Weather
{
    public string WeatherType;

    public Weather(string weatherType)
    {
        WeatherType = weatherType;
    }

    public void ChangeWeather(string newWeather)
    {
        WeatherType = newWeather;
    }
}

class Animal
{
    public string Name;
    public string Species;
    public int Hunger;
    public int Health;

    public Animal(string name, string species, int hunger, int health)
    {
        Name = name;
        Species = species;
        Hunger = hunger;
        Health = health;
    }

    public void Feed()
    {
        Hunger -= 10;
    }

    public void Heal()
    {
        Health += 10;
    }

    public string Produce()
    {
        if (Species == "cow")
        {
            return "milk";
        }
        else if (Species == "chicken")
        {
            return "eggs";
        }
        else
        {
            return "nothing";
        }
    }
}

class Barn
{
    public int Capacity;
    public List<Animal> Animals;

    public Barn(int capacity)
    {
        Capacity = capacity;
        Animals = new List<Animal>();
    }

    public void AddAnimal(Animal animal)
    {
        if (Animals.Count < Capacity)
        {
            Animals.Add(animal);
        }
    }

    public void FeedAnimals()
    {
        foreach (Animal animal in Animals)
        {
            animal.Feed();
        }
    }
}

class Farm
{
    public List<Crop> Crops;
    public Barn Barn;
    public Weather Weather;

    public Farm(Barn barn, Weather weather)
    {
        Crops = new List<Crop>();
        Barn = barn;
        Weather = weather;
    }

    public void PlantCrop(Crop crop)
    {
        Crops.Add(crop);
    }

    public void NextDay()
    {
        foreach (Crop crop in Crops)
        {
            crop.Grow(Weather.WeatherType);
        }
    }

    public void FeedAnimals()
    {
        Barn.FeedAnimals();
    }

    public int HarvestAll()
    {
        int totalYield = 0;

        foreach (Crop crop in Crops)
        {
            totalYield += crop.Harvest();
        }

        return totalYield;
    }
}

class Program
{
    static void Main(string[] args)
    {
        Weather weather = new Weather("sunny");
        Barn barn = new Barn(5);
        Farm farm = new Farm(barn, weather);

        Animal cow = new Animal("Krowa", "cow", 50, 80);
        Animal chicken = new Animal("Kura", "chicken", 30, 90);

        barn.AddAnimal(cow);
        barn.AddAnimal(chicken);

        Crop wheat = new Crop("Wheat", 0, 5, "rainy");
        farm.PlantCrop(wheat);

        farm.NextDay();
        farm.NextDay();
        farm.NextDay();

        farm.FeedAnimals();

        Console.WriteLine("Pogoda: " + farm.Weather.WeatherType);
        Console.WriteLine("Wzrost rośliny: " + wheat.Growth);
        Console.WriteLine("Krowa produkuje: " + cow.Produce());
        Console.WriteLine("Kura produkuje: " + chicken.Produce());
        Console.WriteLine("Zebrany plon: " + farm.HarvestAll());
    }
}

File: zadanie2/python/main.py
#Symulator farmy - 25#
class Farm:
    def __init__(self, barn, weather):
        self.crops = []
        self.barn = barn
        self.weather = weather
    
    def plant_crop(self, crop):
        self.crops.append(crop)

    def next_day(self):
        for crop in self.crops:
            crop.grow(self.weather.weather_type)

    def feed_animals(self):
        self.barn.feed_animals()

    def harvest_all(self):
        total_yield = 0
        for crop in self.crops:
            total_yield += crop.harvest()
        return total_yield
        

class Crop:
    def __init__(self, name, growth, yield_amount, weather_impact):
        self.name = name
        self.growth = growth
        self.yield_amount = yield_amount
        self.weather_impact = weather_impact
    
    def grow(self, weather_type):
        if weather_type == "sunny":
            self.growth += 1
        elif weather_type == "rainy":
            self.growth += 2
        elif weather_type == "stormy":
            self.growth += 0

    def harvest(self):
        if self.growth >= 3:
            return self.yield_amount
        else:
            return 0
    
class Weather:
    def __init__(self, weather_type):
        self.weather_type = weather_type

    def change_weather(self, new_weather):
        self.weather_type = new_weather

class Animal:
    def __init__(self, name, species, hunger, health):
        self.name = name
        self.species = species
        self.hunger = hunger
        self.health = health

    def feed(self):
        self.hunger -= 10
        
    def heal(self):
        self.health +=10

    def produce(self):
        if self.species == "cow":
            return "milk"
        elif self.species == "chicken":
            return "eggs"
        else:
            return None
        
class Barn:
    def __init__(self, capacity):
        self.capacity = capacity
        self.animals = []
    
    def add_animal(self, animal):
        if len(self.animals) < self.capacity:
            self.animals.append(animal)

    def feed_animals(self):
        for animal in self.animals:
            animal.feed()
    
weather = Weather("sunny")
barn = Barn(5)
farm = Farm(barn, weather)

cow = Animal("Krowa", "cow", 50, 80)
chicken = Animal("Kura", "chicken", 30, 90)

barn.add_animal(cow)
barn.add_animal(chicken)

wheat = Crop("Wheat", 0, 5, "rainy")
farm.plant_crop(wheat)

farm.next_day()
farm.feed_animals()

print("Pogoda:", farm.weather.weather_type)
print("Wzrost rośliny:", wheat.growth)
print("Krowa produkuje:", cow.produce())
print("Kura produkuje:", chicken.produce())
print("Zebrany plon:", farm.harvest_all())
